apps/prairielearn/src/lib/base64-util.ts

```diff
@@ -1,37 +1,22 @@
-import { logger } from '@prairielearn/logger';
-
-const atob = (s: string) => String.fromCharCode(...Buffer.from(s, 'base64'));
-const btoa = (s: string) => Buffer.from(s.split('').map((c) => c.charCodeAt(0))).toString('base64');
-
 export function b64EncodeUnicode(str: string) {
   // (1) use encodeURIComponent to get percent-encoded UTF-8
   // (2) convert percent encodings to raw bytes
   // (3) convert raw bytes to Base64
-  try {
-    return btoa(
-      encodeURIComponent(str).replaceAll(/%([0-9A-F]{2})/g, (match, p1) => {
-        return String.fromCharCode(Number.parseInt('0x' + p1, 16));
-      }),
-    );
-  } catch {
-    logger.error(`b64EncodeUnicode: returning empty string because failed to encode ${str}`);
-    return '';
-  }
```

Comment:
Logging and returning an empty string is _very_ strange behavior. This actually bit me while working on agentic question generation. I accidentally passed in a non-base64-encoded string to the decode function and was very surprised when I was getting back an empty string. It wasn't until I looked at the logs that I realized what had happened.

IMO it's better to just throw an error if this fails than somewhat-silently using an empty string.